src/fstar/FStar.Interactive.Ide.Types.fst

```diff
@@ -0,0 +1,337 @@
+(*
+   Copyright 2008-2016  Nikhil Swamy and Microsoft Research
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*)
+
+module FStar.Interactive.Ide.Types
```

Comment:
Refactoring several existing modules to put the types in one place, so that we can easily refer to them in other modules without introducing dependency cycles